app/src/gui/components/authentication/cluster_card.tsx

```diff
@@ -18,318 +18,249 @@
-                listing_id={props.listing_id}
-                current_username={token.parsedToken.username}
-                // TODO should anything happen when the token/username changes?
-                onUpdated={() => {}}
-              />
+                  {tokenInfo?.parsedToken?.name && (
+                    <Typography variant="body2" color="text.secondary">
+                      {tokenInfo.parsedToken.name}
+                    </Typography>
+                  )}
 
-              <LoginButton
-                key={props.listing_id}
-                conductor_url={props.conductor_url}
-                is_refresh={true}
-                label={'add another user'}
-                size={'small'}
-                sx={{my: 1}}
-                startIcon={<PersonAddIcon />}
-              />
-            </React.Fragment>
-          ) : (
-            ''
-          )}
-        </React.Fragment>
+                  {/* Alert Section */}
+                  {!tokenValid && isLoggedIn && (
+                    <Alert severity={'error'}>
+                      This login has expired. Click refresh, below, to login
+                      again.
+                    </Alert>
+                  )}
+
+                  {/* Button Section */}
+                  {!isLoggedIn ? (
+                    <LoginButton
+                      key={props.serverId}
+                      conductor_url={props.conductor_url}
+                      is_refresh={false}
+                      startIcon={<LoginIcon />}
+                    />
+                  ) : (
+                    <Box
+                      sx={{
+                        display: 'flex',
+                        flexWrap: 'wrap',
+                        gap: 1,
+                        '& > button': {
+                          flex: {
+                            // Full width on very small screens
+                            xs: '1 1 100%',
+                            // Auto width on larger screens
+                            sm: '1 1 auto',
+                          },
+                          minWidth: {
+                            // Minimum button width on larger screens
+                            sm: '120px',
+                          },
+                        },
+                      }}
+                    >
+                      {!isActive && (
+                        <Button
+                          size={'small'}
+                          variant={'outlined'}
+                          onClick={() => dispatch(setActiveUser(identity))}
+                          startIcon={<Person2Sharp />}
+                        >
+                          Activate
+                        </Button>
+                      )}
+                      <LoginButton
+                        key={props.serverId}
+                        conductor_url={props.conductor_url}
+                        is_refresh={true}
+                        label={'refresh'}
```

Comment:
Still not sure about 'Refresh' as a label for this button. Maybe ask Elana but it doesn't really tell me why I would click it.  I've never seen a refresh button in any other app I've used.  I think we at least need some text to explain why clicking it might be required sometimes (your login token will last for N days before you need to re-login, click Refresh to extend your login session).

Reply:
Actually, what it does when the token expires makes sense, so it's just when it's not urgent that it's confusing.

Reply:
Only shows when token is expired with error indication and refined wording :)